src/util/db/dbconnectionpooler.cpp

```diff
@@ -28,4 +28,12 @@ DbConnectionPooler::~DbConnectionPooler() {
     }
 }
 
+DbConnectionPooler& DbConnectionPooler::operator=(DbConnectionPooler&& other) {
+    m_pDbConnectionPool = std::move(other.m_pDbConnectionPool);
+    // Move assignment should transfer ownership by invalidating
+    // the other instance.
+    DEBUG_ASSERT(!other);
```

Comment:
This is hard to understand here, the drawback of operator overloads. 
Ow about replace it by a named function like 
DEBUG_ASSERT(!other.initalized());
Or just by 
DEBUG_ASSERT(!other.m_pDbConnectionPool) 
or just remove the Assert, because it is obvious that it works.